lego/apps/surveys/tasks.py

```diff
@@ -0,0 +1,26 @@
+from django.utils import timezone
+from structlog import get_logger
+
+from lego import celery_app
+from lego.apps.events.constants import PRESENT
+from lego.apps.stats.utils import track
+from lego.apps.surveys.models import Submission, Survey
+from lego.apps.surveys.notifications import SurveyNotification
+from lego.utils.tasks import AbakusTask
+
+log = get_logger()
+
+
+@celery_app.task(serializer='json', bind=True, base=AbakusTask)
+def send_survey_mail(self, logger_context=None):
+    self.setup_logger(logger_context)
+
+    surveys = Survey.objects.filter(active_from__lte=timezone.now(), sent=False)
+    for survey in surveys.all():
+        for registration in survey.event.registrations.filter(presence=PRESENT):
+            if not Submission.objects.filter(survey=survey, user=registration.user).exists():
```

Comment:
How about something like this? It will still notify a user several times if they don't have time to answer the survey in-between, but it should at least prevent any "I already answered this survey, why am I getting mail?!?!" messages.

Reply:
Since you're setting `sent = True` it will only run once anyway though, right? That is, it won't send out any extra reminders since after the first time `sent` will be `True`.

I'm not sure if periodic reminders are actually necessary though, since they'll have to answer the survey before their next event regardless. And in that case, you don't need the `Submission.objects.filter` logic, since no one will be able to answer the survey before it's sent out.

Reply:
I was thinking of this as a fix to the issue you described where the system could crash after some users had received a mail but before `sent` was set to `True`. Like I said, this would only help if the users who receive mails before the crash have time to answer the survey before its sent again, but that's better than nothing?

Reply:
Ah, I see - I misunderstood then. I don't think that's a problem you need to solve, if it somehow crashes in the middle of your task (which would involve something crazy like the database doing down) I don't think it really matters if people get an extra email once.

Reply:
Alright, cool! Will drop this check then